extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -76,6 +76,15 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		if (bookPathToTrust) {
 			let trustChanged = this._bookTrustManager.setBookAsTrusted(bookPathToTrust);
 
+			// update trust state of opened items
+			if (azdata.nb.notebookDocuments && azdata.nb.notebookDocuments.length) {
+				azdata.nb.notebookDocuments.forEach(document => {
+					if (this._bookTrustManager.isNotebookTrustedByDefault(document.uri.fsPath)) {
```

Comment:
You should trust the book only when it's not already trusted. And I agree with Charles, we should only do this for notebooks within the book that was just trusted, no point in looping through all the open notebooks. May be you can filter the open notebooks that are under this book first?

Reply:
# resolved